x/ecocredit/marketplace/keeper/utils.go

```diff
@@ -192,25 +198,109 @@ func (k Keeper) fillOrder(ctx context.Context, params fillOrderParams) error {
 		}
 	}
 
-	// update buyer balance with new tradable or retired amount
+	// update buyer credit balance with new tradable or retired amount
 	if err = k.baseStore.BatchBalanceTable().Save(ctx, buyerBal); err != nil {
 		return err
 	}
 
-	// send total cost from buyer account to seller account
-	return k.bankKeeper.SendCoins(sdkCtx, params.buyerAcc, params.sellOrder.Seller, sdk.NewCoins(params.totalCost))
+	// calculate seller fee = subtotal * seller percentage fee
+	sellerFee, err := getSellerFee(params.subTotalCost, params.feeParams)
+	if err != nil {
+		return err
+	}
+
+	// calculate total fee = buyer fee + seller fee
+	totalFee, err := params.buyerFee.Add(sellerFee)
+	if err != nil {
+		return err
+	}
+
+	// if total fee > 0, then transfer total fee from buyer account to fee pool
```

Comment:
This looks good to me, just a question on how the fees/costs are moving around.

As implemented here a successful `MsgBuyDirect` in a TX with quantity 10, cost 10uregen and both a buyer fee 0.1 and seller fee 0.05, will result in two separate `transfer` events in the message :
1. `transfer` for all total fees (15uregen) from the buyer to the fee pool
2. `transfer` for seller payment (95uregen) from the buyer to the seller

This makes sense and the math adds up. But I wonder if it could lead to some confusion because from the resulting tx it looks like the buyer is paying all of the fees. I'm not sure how important it is too keep this detailed of a record on the TX or if it is much less efficient, but it seems like it could be useful for helpting to calculate total fees spent by buyer or seller (maybe for tax purposes?)

An example:
1. `transfer` for only buyer fees (10uregen) from the buyer to the fee pool
2. `transfer` for seller payment subtotal cost (100uregen) from the buyer to the seller
3. `transfer` for only seller fees (5uregen) from the seller to the fee pool

Reply:
Personally, I would stick to the side of efficiency here and not introduce unnecessary state changes because of downstream accounting. If accounting is the consideration, I would rather deal with that by adding attributes to events, maybe enhancing `EventBuyDirect`?

Reply:
Yeah, it might make sense to add that in `EventBuyDirect` then. I'm just assuming this would be useful for accounting, maybe we can validate that from a business use-case.

Reply:
I agree for accounting maybe the event makes more sense

Reply:
@paul121 @JeancarloBarrios added seller and buyer fees paid to `EventBuyDirect`, see https://github.com/regen-network/regen-ledger/pull/2154/commits/b1a7ad233d9a64df09464a547fc7b170fb2ec596. Does that address this sufficiently?